### ISSUE TITLE
error when get _responses

### ISSUE THREAD

**Issue by gitbobo** [cloudyr/limer#30]
> get_session_key()
[1] "ddrvaxfczqv9i5bhf4h34ukq5h86f4jt"
> responses <- get_responses(913762)
Error: lexical error: invalid char in json text.
                                       A language code must be supplie
                     (right here) ------^

could you pls tell me what's the problem may be, thanks!

**Comment by r0berts** [comment]
I got the same error and then I looked at survey permissions - turned out that the user did not have survey permissions, once I added them I got data. Strange thing however was that I got all fields in one column. Troubleshooting that now.

**Comment by Jan-E** [comment]
@r0berts 
>Strange thing however was that I got all fields in one column. Troubleshooting that now.

Could that be related to
https://github.com/cloudyr/limer/commit/d7c0c3ecf4e2393045711622589165e75c3b5f4d

**Comment by r0berts** [comment]
@Jan-E That is right - it was. It is because I had an older version (Version 2.67.1+170626) where the field separator was different. I solved this by installing a version pre-29th April. `install_github("cloudyr/limer", ref = "f0e2acced9178133f56f09f40ba690d6787eda64")` - I will update the installation some time soon, but presently I just needed the exporting to work.

**Comment by Jan-E** [comment]
@r0berts We had it the other way around. Limesurvey 3.12.1, but a limer that was too old. So we just had to update limer. I will see if I can patch it do do a version-specific base64_to_df.

**Comment by Jan-E** [comment]
The PR seems to be already there: https://github.com/cloudyr/limer/pull/42

**Comment by r0berts** [comment]
@Jan-E this is good, perhaps to have an option to supply version or other switch. If need be, I could do some work on documenting limer, so that it would be easier for newcomers to figure this out. I could do something like a manpage  from a begginner's viewpoint, if you'd like - maybe to supplement the readme file?

**Comment by Jan-E** [comment]
I am already experimenting to get the version number:
```
get_session_key()
lsversion<-call_limer(method="get_site_settings",params=list(sSetttingName="versionnumber"))
lsmajor<-substr(lsversion, 1, 1)
print(lsmajor)
```
In my case lsversion becomes "3.12.1" and lsmajor "3". With this it should be possible to restrict https://github.com/cloudyr/limer/commit/d7c0c3ecf4e2393045711622589165e75c3b5f4d to versions > 2.

**Comment by Jan-E** [comment]
@r0berts Could you test
`devtools::install_github("toolsforresearch/limer", username = NULL, ref = "ls_v2_export_responses")`

https://github.com/toolsforresearch/limer/commit/967873d45f6fcb52a5dbbc60ba56fb81255bb00e handles LS v2 the old way and LS v3 with a sep = ";". I tested it for Limesurvey 3.12.1.

Can you confirm it works in LS v2? Then I will create e new Pull Request.

**Comment by r0berts** [comment]
It seems to work. There are error messages not present in main branch - which might be confusing:
```
Error in if (lsmajor < 3) { : missing value where TRUE/FALSE needed
In addition: Warning message:
In base64_to_df(unlist(results)) : NAs introduced by coercion
```
But apart from that looks good.

**Comment by Jan-E** [comment]
Looks like lsmajor is not set. Could you run this in a R prompt, after entering your lime_api optiions:
```
library(limer)
get_session_key()
lsversion<-call_limer(method="get_site_settings",params=list(sSetttingName="versionnumber"))
print(lsversion)
lsmajor<-as.numeric(substr(lsversion, 1, 1))
print(lsmajor)
```

**Comment by r0berts** [comment]
Yes, lsmajor is returned as list with this content:
```
> str(lsversion)
List of 1
 $ status: chr "Invalid setting"
```

**Comment by Jan-E** [comment]
@r0berts 
Ouch. We also had a script with LS 2 and a non-superadmin RPC2 user. Hence, it failed. I now tried another method: test if the 'copy_survey' exists. copy_survey was introduced in LS 3.0.

https://github.com/toolsforresearch/limer/commit/d92be6f52ad50e1dcffdca9d7af4fbe3a93931d7?diff=unified contains the final function:
```
base64_to_df <- function(x) {
  raw_csv <- rawToChar(base64enc::base64decode(x))

  have_copy_survey <- call_limer(method="copy_survey")
  # returns NULL in LS2
  # returns array(status => "Copy failed", error => "No survey ID has been provided. Cannot copy survey") in LS3
  if (is.null(have_copy_survey)) {
    return(read.csv(textConnection(raw_csv), stringsAsFactors = FALSE))
  } else {
    return(read.csv(textConnection(raw_csv), stringsAsFactors = FALSE, sep = ";"))
  }
}
```
I confirmed that this works in LS Version 2.6.4-lts with a non-superadmin account. Could you once again test the patch?

```
devtools::install_github("toolsforresearch/limer", username = NULL, ref = "ls_v2_export_responses_version_2")
```
Thanks.

**Comment by Jan-E** [comment]
Yes, but they are quite convinced, that you spell it that way
https://github.com/LimeSurvey/LimeSurvey/blob/master/application/helpers/remotecontrol/remotecontrol_handle.php#L95

**Comment by r0berts** [comment]
@Jan-E 

Hi Jan, sorry I was away. I tested today. Your version works fine on my 2.67 - it gets data perfectly alright. Then I tested it on the latest version of LS running on my laptop - it fails with this error:

``` 
 > initData <- get_responses(limeSurveyNumber)  
Error: lexical error: invalid char in json text.  
                                       <!DOCTYPE html> <html lang="en"  
                     (right here) ------
```

I also tested the latest version with the version from standard repo https://github.com/cloudyr/limer and that works fine on the latest.

**Comment by tammoterhark** [comment]
@Jan-E Would it be an idea to have two versions of this piece of software: limer and limer3, to avoid this confusion.

And to persuade the LS developers to make it possible to ask the API for a version number?

Using LS3.14 I ran into this problem again.

**Comment by digisus** [comment]
Hi, got the same error like the OP: `invalid multibyte string 1` using Limer 6.13.2 and RStudio 2025.05.0-496.

Sadly I do not understand @r0berts initial answer with the survey permissions: which user and which survey permissions?

I have only one LS user and would assume it has all permissions...? Thanks.
